### DISCUSSION TITLE
Errors within Extensions are not caught

### DISCUSSION THREAD

**Discussion by cynova** [directus/directus#6641] [Feature Requests]
### Describe the Bug

When an extension throws an error (for example a created hook or a custom route), the whole process crashes.
In my opinion errors should be handled as follows:
- In .before hook: fail the whole request (don't write to DB) and return an internal server error
- In normal hook: do nothing (except maybe log)
- In custom route: return internal server error

Also, the Process crashes if a module extension isn't well formed (no export). This seems reasonable though.

### To Reproduce

`module.exports = function registerEndpoint(
    router,
) {
    router.get(
        '/',
        ()=>{throw new Error();}
    );
};
`

### What version of Directus are you using?

9.0.0 rc-83

### What version of Node.js are you using?

-

### What database are you using?

mariadb

### What browser are you using?

chrome

### What operating system are you using?

macOS

### How are you deploying Directus?

Docker (official image)

**Comment by rijkvanzanten** [comment]
We'll have to figure out how much hand-holding/magic we wanna do on the extensions side. API extensions now are a "do whatever you want at your own risk" approach, which give the most amount of flexibility, but also have a couple of places where it's easy to get yourself into trouble.

@nickrum ideally, we wrap the custom endpoints in the same `asyncHandler` util we use for the main routes, which will take care of the proper error handling to the client

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
